File: data/Pandora.Data/Models/TerraformModelToModelMappingDefinition.cs
// Copyright (c) HashiCorp, Inc.
// SPDX-License-Identifier: MPL-2.0

namespace Pandora.Data.Models;

public class TerraformModelToModelMappingDefinition
{
    /// <summary>
    /// SchemaModelName is the name of the Schema Model that there's a mapping between.
    /// </summary>
    public string SchemaModelName { get; set; }

    /// <summary>
    /// SdkModelName is the name of the Sdk Model that there's a mapping between.
    /// </summary>
    public string SdkModelName { get; set; }
}

File: tools/data-api-sdk/v1/models/terraform_model_to_model_mapping_definition.go
// Copyright IBM Corp. 2021, 2025
// SPDX-License-Identifier: MPL-2.0

package models

// TerraformModelToModelMappingDefinition defines XXX.
// This is used to output both the `map{TerraformSchemaModelName}To{SDKModelName}` and
// `map{SDKModelName}To{TerraformSchemaModelName}` functions.
type TerraformModelToModelMappingDefinition struct {
	// SDKModelName specifies the name of the SDKModel which should be mapped to/from the
	// TerraformSchemaModel named in TerraformSchemaModelName.
	SDKModelName string `json:"sdkModelName"`

	// TerraformSchemaModelName specifies the name of the TerraformSchemaModel which should
	// be mapped to/from the SDKModel named in SDKModelName.
	TerraformSchemaModelName string `json:"schemaModelName"`
}
